client/client.go

```diff
@@ -45,6 +46,11 @@ type client struct {
 	//metadataLock sync.RWMutex
 }
 
+// Returns the URL the client connects to
+func (c client) GetURL() string {
```

Comment:
in go getters shall not contain `Get` by convention (For RPCs its ok as they are not getters, but network calls mapping to rpc calls).
```suggestion
func (c client) URL() string {
```